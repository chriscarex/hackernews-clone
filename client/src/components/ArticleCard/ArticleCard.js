import React from 'react'
import PropTypes from 'prop-types'
import { Card, Icon } from 'semantic-ui-react'
import Moment from 'react-moment'
import moment from 'moment'

export const ArticleCard = ({
  article: {
    by,
    score,
    time,
    title,
    url
  }
}) => (
  <a
    href={url}
    target="_blank"
    rel="noopener nofollow noreferrer"
    className="article-card"
    data-cy="article-card"
  >
    <Card className="article-card-container">
      <Card.Content className="article-card-title">{title}</Card.Content>
      <Card.Content extra className="article-card-meta">
        <div className="article-card-meta-item">
          <Icon name="user" /> {by}
        </div>

        <div className="article-card-meta-item">
          <Icon name="certificate" /> {score}
        </div>

        <div className="article-card-item">
          <Icon name="clock" /> <Moment fromNow>{moment.unix(time).format('YYYY-MM-DD HH:mm')}</Moment>
        </div>
      </Card.Content>
      {url
          && (
          <Card.Content>
            <Icon name="globe" /> {url && url.length > 80 ? `${url.substring(0, 80)}...` : url}
          </Card.Content>
        )}
    </Card>
  </a>
)

ArticleCard.propTypes = {
  // index: PropTypes.number.isRequired,
  article: PropTypes.shape({
    id: PropTypes.number,
    score: PropTypes.number,
    by: PropTypes.string,
    time: PropTypes.number,
    title: PropTypes.string,
    // descendants: PropTypes.string,
    url: PropTypes.string,
  }).isRequired,
}

export default ArticleCard // withRouter(ArticleCard)
