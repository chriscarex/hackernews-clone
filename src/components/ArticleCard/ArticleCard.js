import React from 'react'
import PropTypes from 'prop-types'
import { Card, Icon } from 'semantic-ui-react'
import Moment from 'react-moment'
import moment from 'moment'

export const ArticleCard = ({
  article: {
    by,
    score,
    time,
    title,
    url
  }
}) => (
  <a
    href={url}
    target="_blank"
    rel="noopener nofollow noreferrer"
    className="article-card"
    data-cy="article-card"
  >
    <Card className="article-card-container">
      <Card.Content className="article-card-title">{title}</Card.Content>
      <Card.Content extra className="article-card-meta">
        <div className="article-card-meta-item">
          <Icon name="user" /> {by}
        </div>

        <div className="article-card-meta-item">
          <Icon name="certificate" /> {score}
        </div>

        <div className="article-card-item">
          <Icon name="clock" /> <Moment fromNow>{moment.unix(time).format('YYYY-MM-DD HH:mm')}</Moment>
        </div>
      </Card.Content>
      {url
          && (
          <Card.Content className="article-card-item">
            <Icon name="globe" /> {url && url.length > 80 ? `${url.substring(0, 80)}...` : url}
          </Card.Content>
        )}
    </Card>
  </a>
)

ArticleCard.propTypes = {
  article: PropTypes.shape({
    id: PropTypes.number,
    score: PropTypes.number,
    by: PropTypes.string,
    time: PropTypes.number,
    title: PropTypes.string,
    url: PropTypes.string,
  }).isRequired,
}

export default ArticleCard
