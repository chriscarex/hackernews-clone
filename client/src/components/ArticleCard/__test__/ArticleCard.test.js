import React from 'react'
// import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import moment from 'moment'
import { ArticleCard } from '../ArticleCard'

// const setup = () => {
//   const props = {
//     article: {
//       by: 'authry',
//       score: 2,
//       time: '10000000',
//       title: 'Title',
//       url: 'http://test.com'
//     }
//   }
//
//   const component = shallow(<ArticleCard {...props} />)
//
//   return {
//     component
//   }
// }

describe('ArticleCard', () => {
  it('should render correctly', () => {
    const article = {
      by: 'author',
      score: 2,
      time: '10000000',
      title: 'Title',
      url: 'http://test.com'
    }

    const tree = renderer
    .create(<ArticleCard article={article} />)
    .toJSON()

    expect(tree).toMatchSnapshot(`
<a
href='http://test.com'
target="_blank"
rel="noopener nofollow noreferrer"
className="article-card"
data-cy="article-card"
>
  <Card className="article-card-container">
    <Card.Content className="article-card-title">{title}</Card.Content>
    <Card.Content extra className="article-card-meta">
      <div className="article-card-meta-item">
        <Icon name="user" /> author
      </div>

      <div className="article-card-meta-item">
        <Icon name="certificate" /> 2
      </div>

      <div className="article-card-item">
        <Icon name="clock" />
        <time
           dateTime={"1970-04-26T17:46:00.000Z"}
        >
         50 years ago
        </time>
      </div>
    </Card.Content>
    <Card.Content>
    <Icon name="globe" /> http://test.com
    </Card.Content>
  </Card>
</a>
`)
  })
})
