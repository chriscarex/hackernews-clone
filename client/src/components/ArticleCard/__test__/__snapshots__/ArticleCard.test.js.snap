// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleCard should render correctly: 
<a
href='http://test.com'
target="_blank"
rel="noopener nofollow noreferrer"
className="article-card"
data-cy="article-card"
>
  <Card className="article-card-container">
    <Card.Content className="article-card-title">{title}</Card.Content>
    <Card.Content extra className="article-card-meta">
      <div className="article-card-meta-item">
        <Icon name="user" /> author
      </div>

      <div className="article-card-meta-item">
        <Icon name="certificate" /> 2
      </div>

      <div className="article-card-item">
        <Icon name="clock" />
        <time
           dateTime={"1970-04-26T17:46:00.000Z"}
        >
         50 years ago
        </time>
      </div>
    </Card.Content>
    <Card.Content>
    <Icon name="globe" /> http://test.com
    </Card.Content>
  </Card>
</a>
 1`] = `
<a
  className="article-card"
  data-cy="article-card"
  href="http://test.com"
  rel="noopener nofollow noreferrer"
  target="_blank"
>
  <div
    className="ui card article-card-container"
    onClick={[Function]}
  >
    <div
      className="content article-card-title"
    >
      Title
    </div>
    <div
      className="extra content article-card-meta"
    >
      <div
        className="article-card-meta-item"
      >
        <i
          aria-hidden="true"
          className="user icon"
          onClick={[Function]}
        />
         
        author
      </div>
      <div
        className="article-card-meta-item"
      >
        <i
          aria-hidden="true"
          className="certificate icon"
          onClick={[Function]}
        />
         
        2
      </div>
      <div
        className="article-card-item"
      >
        <i
          aria-hidden="true"
          className="clock icon"
          onClick={[Function]}
        />
         
        <time
          dateTime={"1970-04-26T17:46:00.000Z"}
        >
          50 years ago
        </time>
      </div>
    </div>
    <div
      className="content"
    >
      <i
        aria-hidden="true"
        className="globe icon"
        onClick={[Function]}
      />
       
      http://test.com
    </div>
  </div>
</a>
`;
